### PR TITLE
Feature/server integration

### PR DESCRIPTION
**Pull request by Taloreff-hls**
### Integrated Routes:
- **`GET /pokemons/random?userId={userId}`**
  - Fetches a random Pokémon for the user.
  - The userId is passed as a query parameter to ensure the fetched Pokémon is not already owned by the user.

- **`POST /pokemons`**
  - Fetches Pokémon data with filtering, sorting, and pagination.
  - The backend supports sorting (`sort_by`, `sort_order`), pagination (`page`, `limit`), and search filters (e.g., name, attack, health).
  - Depending on the provided userId, the request fetches either all available Pokémon or the user’s owned Pokémon.

- **`POST /pokemons/users/:id/catch`**
  - Allows users to catch a Pokémon and add it to their collection.
  - The request includes the userId and the `pokemonId` of the captured Pokémon.

### Key Changes:
- **Frontend Hooks Integration:**
  - Hooks (`usePokemonTableData`, `usePokemonGridData`, and `useUserPokemons`) now communicate with the backend to fetch Pokémon data, with support for user ownership flags.
  - Pokémon data is extended with the `belongsToUser` flag, indicating whether a Pokémon is owned by the user, and this is reflected in the UI.
  - Pagination, sorting, and filtering are handled via server-side queries.

- **Ownership Handling:**
  - Pokémon data includes the `belongsToUser` field to indicate ownership, which is determined by checking if a fetched Pokémon exists in the user’s collection.

### CODE REVIEW

File: src/hooks/fightArenaHooks.tsx [hunk 1]
```diff
@@ -186,6 +186,20 @@ export const useBattleLogic = (
           });
         } else {
           setBattleResult(FightResultEnum.CAUGHT);
+
+          try {
+            const userId = "02fea148-e9dc-4cae-89aa-8db50df0dd48"; // Replace with actual user ID
```
Comment (DinFridman) [inline src/hooks/fightArenaHooks.tsx:191]: why its hardcoded?

File: src/hooks/pokemonDataHooks.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  sort_by,
+  sort_order,
+  searchValue,
+  selectedCtg,
+}: PokemonQueryParams) => {
+  return useInfiniteQuery(
+    ["infinitePokemonData", selectedCtg, sort_by, sort_order, searchValue],
+    ({ pageParam = 0 }) =>
+      pokemonService.getPokemons(
+        selectedCtg === 1 ? "02fea148-e9dc-4cae-89aa-8db50df0dd48" : undefined,
```
Comment (DinFridman) [inline src/hooks/pokemonDataHooks.tsx:66]: is this the hardcoded id?

**Review by DinFridman — CHANGES_REQUESTED** [review]

File: src/hooks/fightArenaHooks.tsx (hunk 1 above)
Comment (Taloreff-hls) [reply src/hooks/fightArenaHooks.tsx:191]: About the hardcoded ID, it was to prepare for authentication. After merging, I will need to integrate the user with the service that fetches the data using the ID of the logged-in user.

File: src/hooks/pokemonDataHooks.tsx (hunk 2 above)
Comment (Taloreff-hls) [reply src/hooks/pokemonDataHooks.tsx:66]: About the hardcoded ID, it was to prepare for authentication. After merging, I will need to integrate the user with the service that fetches the data using the ID of the logged-in user.

**Pull request merged**